src/coreclr/jit/async.cpp

```diff
@@ -1100,6 +1098,53 @@ void Async2Transformation::Transform(
     m_resumptionBBs.push_back(resumeBB);
 }
 
+GenTreeCall* Async2Transformation::CreateAllocContinuationCall(AsyncLiveness& life, GenTree* nextContinuation, unsigned gcRefsCount, unsigned dataSize)
+{
+    GenTree* gcRefsCountNode = m_comp->gtNewIconNode((ssize_t)gcRefsCount, TYP_I_IMPL);
+    GenTree* dataSizeNode = m_comp->gtNewIconNode((ssize_t)dataSize, TYP_I_IMPL);
+    // If VM requests that we report the method handle, or if we have a shared generic context method handle
+    // that is live here, then we need to call a different helper.
+    GenTree* methodHandleArg = nullptr;
+    GenTree* classHandleArg = nullptr;
+    if (((m_comp->info.compMethodInfo->options & CORINFO_GENERICS_CTXT_FROM_METHODDESC) != 0) && life.IsLive(m_comp->info.compTypeCtxtArg))
+    {
+        methodHandleArg = m_comp->gtNewLclvNode(m_comp->info.compTypeCtxtArg, TYP_I_IMPL);
+    }
+    else if (((m_comp->info.compMethodInfo->options & CORINFO_GENERICS_CTXT_FROM_METHODTABLE) != 0) && life.IsLive(m_comp->info.compTypeCtxtArg))
+    {
+        // TODO: Do we also need to keep the method handle's loader allocator
+        // alive here in case the shared generic method is inside a collectible
+        // ALC?
```

Comment:
@davidwrighton @VSadov Any idea? Is it sufficient to keep the `LoaderAllocator` associated with the `MethodTable*` generic context alive here? I see that `MethodDesc::GetLoaderAllocator` gets it via `GetLoaderModule()->GetLoaderAllocator()`, but `MethodDesc::GetLoaderModule()` seemingly may not always match `GetMethodTable()->GetLoaderModule()` based on this code:

https://github.com/dotnet/runtime/blob/0170d78578a04206542f32e4db3eaa88a5384973/src/coreclr/vm/method.cpp#L1479-L1492

Reply:
> Is it sufficient to keep the LoaderAllocator associated with the MethodTable* generic context alive here?

I do not think so. You can have `void MyMethod<T>()` instantiated as `void MyMethod<UnloadableType>`. The MethodTable won't know anything about the UnloadableType.

Reply:
Ignore my comment. I have missed that this path is for the MethodTable* generic context case.

It is sufficient to keep the LoaderAllocator associated with the MethodTable* generic context alive here.